url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/template/builder/beans/IndexedPropertyDescriptor.java

```
@@ -0,0 +1,383 @@
+            throw new IntrospectionException(Messages.getString("custom.beans.5A")); //$NON-NLS-1$
+        }
+
+        // Set the indexed property type if not already set, confirm validity if
+        // it is.
+        if (this.indexedGetter == null) {
+            indexedPropertyType = indexedReadType;
+        } else {
+            if (indexedPropertyType != indexedReadType) {
+                // custom.beans.5A=Indexed read method is not compatible with indexed write method
+                throw new IntrospectionException(Messages.getString("custom.beans.5A")); //$NON-NLS-1$
+            }
+        }
+
+        // Set the indexed getter
+        this.indexedGetter = indexGetter;
+    }
+
+    private void setIndexedWriteMethod(Class<?> beanClass, String indexedSetterName)
+            throws IntrospectionException {
+        Method setter = null;
+        try {
+            setter = beanClass.getMethod(indexedSetterName, new Class[] {
+                    Integer.TYPE, getPropertyType().getComponentType() });
+        } catch (SecurityException e) {
+            // custom.beans.5C=Security violation accessing indexed write method
+            throw new IntrospectionException(Messages.getString("custom.beans.5C")); //$NON-NLS-1$
+        } catch (NoSuchMethodException e) {
+            // custom.beans.5D=No such indexed write method
+            throw new IntrospectionException(Messages.getString("custom.beans.5D")); //$NON-NLS-1$
+        }
+        internalSetIndexedWriteMethod(setter, true);
+    }
+
+    private void setIndexedWriteMethod(Class<?> beanClass,
+            String indexedSetterName, Class<?> argType)
+            throws IntrospectionException {
+        try {
+            Method setter = beanClass.getMethod(indexedSetterName, new Class[] {
+                    Integer.TYPE, argType });
+            internalSetIndexedWriteMethod(setter, true);
+        } catch (NoSuchMethodException exception) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 326.